generator/osm_source.cpp

```diff
@@ -120,7 +120,7 @@ void ProcessOsmElementsFromXML(SourceReader & stream, std::function<void(OsmElem
   while (processorOsmElementsFromXml.TryRead(element))
   {
     processor(std::move(element));
-    element.Clear();
+    element = {};
```

Comment:
What was wrong with Clear?

Reply:
Hm, need to check what Clear actually does, but empty initialization is definitely better than some Clear after move.

Reply:
Why is it _definitely_ better? What are the pros and cons?

Reply:
It strongly depends on what Clear actually does. Clear call can _depend_ on some inner state, which is undefined after move. Clear call can reset/initialize not all class inner fields.

Reply:
Then _Clear_ should either be _fixed_ to provide expected behavior or it should be _removed_ from the class.

Reply:
Hm, ok. Then write comment here.

Reply:
> 
element.Clear(); should run faster because the container capacity does not change and we reduce a number of memory reallocations in the future. I would ask for performance tests. (The secret is that it is important at this place.)

Which containers do you mean here? These ones? https://github.com/organicmaps/organicmaps/blob/82f71e159ea5198ce3a82471a7129f41e816e6e2/generator/osm_element.hpp#L167

I am not C++ expert, but it seems that after moving they become invalid and we shouldn't call `clear()` on them here https://github.com/organicmaps/organicmaps/blob/82f71e159ea5198ce3a82471a7129f41e816e6e2/generator/osm_element.hpp#L91

Should we? 

In other words:
1. We move `OsmElement` in this place where I fixed it (i.e. its internal containers are in invalid state now)
2. Then call `OsmElement::Clear`, which in its turn calls `clear()` on `std::vector/std::string` which are now in invalid state after moving

WDYT?

Reply:
BTW, not sure (probably it's not specified) but vector = {} should keep the capacity.

Reply:
Moreover, if we move the vector, there is no sense think about capacity of previous vector. The _capacity_ buffer definitely will be also moved.

Reply:
@SiarheiFedartsou the object after move is not _invalid_. It is `valid but in unspecified state`. So calling `clear()` is perfectly fine, while calling pop_back() is UB.

https://stackoverflow.com/a/9168917

Reply:
hm, interesting, didn't know that. BUT I doubt it somehow helps with allocations(I assume `clear` will allocate new buffer in this case) + if in the case like this we can avoid "use after move", why don't we do that to avoid bugs in the future? after all who knows how we may want to modify this `Clear` method in the future?

Reply:
Well again, = {} is better and more _code-clear_ here than "Clear".

Reply:
When I wrote this, I also knew about  = {}. I did performance measurements, you can try it too. The choice is yours:)

Reply:
Benchmark shows that Clear is faster: https://godbolt.org/z/3TE7WPT4h

I vote for the faster code.

Reply:
@biodranik vector is not moved before `clear` in this benchmark - I bet it should be no difference if you start moving it (at least I cannot imagine what magic clear can do to be faster on moved vector - memory is already moved and it has to allocate new buffer).

Unfortunately I am not near my laptop right now, but will try to “play” with this benchmark later.

Reply:
Hm, I also can't realize what can be "slow" here. I suspect that correct (with move) benchmark will have bigger threshold error than actual difference between clear or default initialization.

Reply:
But I don't mind keep Clear here, write comment and merge this PR at last.